sdk/src/client/api/mana.rs

```diff
@@ -0,0 +1,81 @@
+// Copyright 2023 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    client::Client,
+    types::block::{
+        protocol::{lower_bits, multiplication_and_shift, upper_bits},
+        slot::SlotIndex,
+    },
+};
+
+impl Client {
```

Comment:
I wouldn't put this in the client. All mana calculations should be in the types module, if possible in the mana module.

Reply:
The problem is the necessity of the epochs iter. We need the client to do that calculation anyway

Reply:
Fixed